packages/integration-karma/test/synthetic-shadow/inner-outer-text/inner-outer-text.spec.js

```diff
@@ -0,0 +1,150 @@
+
+            expect(testCase.innerText).toBe(`Chose a pet:`);
+        });
+
+        it('should not collect text from textarea', () => {
+            const testCase = elm.shadowRoot.querySelector('.textarea-testcase');
+
+            expect(testCase.innerText).toBe(`Tell us your story:`);
+        });
+
+        // datalist is supported on safari >= 12
+        if (!process.env.COMPAT) {
+            it('should not collect text from datalist', () => {
+                const testCase = elm.shadowRoot.querySelector('.datalist-case');
+
+                expect(testCase.innerText).toBe(`Choose a flavor:`);
+            });
+        }
+
+        it('should not go inside custom element shadow', () => {
+            const testElement = elm.shadowRoot.querySelector('.without-slotted-content');
+
+            expect(testElement.innerText).toBe('first text\nsecond text');
+        });
+
+        it('should process custom elements light dom', () => {
+            const testElement = elm.shadowRoot.querySelector('.with-slotted-content');
+
+            expect(testElement.innerText).toBe('first text\n\nslotted element\n\nsecond text');
+        });
+
+        it('should process custom elements light dom across multiple shadows', () => {
+            const testElement = elm.shadowRoot.querySelector('.with-slotted-content-2-levels');
+
+            expect(testElement.innerText).toBe('first text\n\nslotted element\n\nsecond text');
+        });
+    });
+
+    const outerTextDescriptor = Object.getOwnPropertyDescriptor(HTMLElement.prototype, 'outerText');
+
+    // Firefox does not have outerText.
+    if (outerTextDescriptor) {
+        describe('outerText', () => {
+            const elm = createElement('x-container', { is: Container });
+            document.body.appendChild(elm);
+
+            return Promise.resolve().then(() => {
+                it('should not go inside custom element shadow', () => {
+                    const testElement = elm.shadowRoot.querySelector('.without-slotted-content');
+
+                    expect(testElement.outerText).toBe('first text\nsecond text');
+                });
+
+                it('should process custom elements light dom', () => {
+                    const testElement = elm.shadowRoot.querySelector('.with-slotted-content');
+
+                    expect(testElement.outerText).toBe(
+                        'first text\n\nslotted element\n\nsecond text'
+                    );
+                });
+
+                it('should process custom elements light dom across multiple shadows', () => {
+                    const testElement = elm.shadowRoot.querySelector(
+                        '.with-slotted-content-2-levels'
+                    );
+
+                    expect(testElement.outerText).toBe(
+                        'first text\n\nslotted element\n\nsecond text'
+                    );
+                });
+            });
+        });
```

Comment:
Lets add a test to confirm the setter is working as expected:
1. Setting innerText of a host element should
   i. Keep the shadow tree intact
   ii. set only the light dom to the new content
   iii. the light dom is correctly slotted into the shadow tree of the host(in synthetic-mode especially)